### PR TITLE
skip recursive lookup value for mustache template

### PR DESCRIPTION
**Pull request by OsmondJiang**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/108774

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4932)

### CODE REVIEW

**Comment by OsmondJiang** [comment]
Seems we can't simply skip recursive look up for all, otherwise it can't support below case which is used in many places:

image

@yufeih  @mingwli

File: src/docfx/template/MustacheTemplate.cs [hunk 1]
```diff
@@ -52,14 +51,15 @@ private static RendererSettingsBuilder UseJson(RendererSettingsBuilder settings)
 
             return settings.AddValueGetter(typeof(JObject), GetJObjectValue)
                            .AddTruthyCheck<JObject>(value => value != null)
+                           .AddTruthyCheck<JValue>(value => value.Type == JTokenType.Null)
```
Comment (yufeih) [inline src/docfx/template/MustacheTemplate.cs]: add a test case:
```suggestion
                           .AddTruthyCheck<JValue>(value => value.Type != JTokenType.Null)
```

Comment (OsmondJiang) [reply src/docfx/template/MustacheTemplate.cs]: oooopsss

**Comment by OsmondJiang** [comment]
the diffs are expected

**Pull request merged**